### PR TITLE
Include break time in timings

### LINKED ISSUE #437: Schedule doesn't count time for breaks
the yaml headers have categories for teaching, exercises, and break time. However, the schedule rendered in the instructor view of the Setup page doesn't seem to count time for breaks, which makes it hard to translate into a local time schedule.

@zkamvar should we fix this by changing the yaml headers in those episodes, or is this something that can be changed in the way the schedule is rendered?

### PR DESCRIPTION
**Pull request by bencomp**
If my naive change works, this fixes #437. However, I did not test this!

### CODE REVIEW

File: R/get_syllabus.R
Comment (zkamvar) [inline R/get_syllabus.R:90]: This will set the default break time to zero so that we don't throw any warnings or add extra time to the schedule for lessons that do not have the explicit break yaml item.

```suggestion
  times <- c(coerce_integer(yaml$teaching), coerce_integer(yaml$exercises), coerce_integer(yaml$break, 0L))
```

**Review by zkamvar — CHANGES_REQUESTED** [review]
Thank you for this, @bencomp!

This is failing in part due to #457, but will be fixed once we merge in #458. The other reason why it would fail is because `coerce_integer()` sets a default of `-5` (in R, bare numbers are usually floats and integers are written with an "L" following them to indicate a "long" integer, though that's not necessarily true anymore.... it's weird, I know) so that we can find the negative numbers and report missing items. 

Once the suggestion is merged, it will pass checks and I will add a new test here to check that adding a break item will work. Once that happens, I will release it. 
https://github.com/carpentries/sandpaper/blob/6989c39f4482b51c22c8ceb38146d08ed8ca4069/tests/testthat/test-get_syllabus.R#L35-L54

PS, I promise I will get to your other PRs 😬

File: R/get_syllabus.R
Comment (zkamvar) [reply R/get_syllabus.R:90]: For reference, the default time was added in version 0.11.6: https://github.com/carpentries/sandpaper/pull/396

**Comment by bencomp** [comment]
Thanks for your comments and help, @zkamvar! I had looked at the tests, but don't fully understand them yet – do they run in order and not reset to a default state between tests, or is there more magic going on with these unit tests?

**Comment by bencomp** [comment]
> PS, I promise I will get to your other PRs 😬

No worries :)

**Comment by bencomp** [comment]
As the code highlighting suggested, `break` is a keyword in R. That means I cannot reference `my_yaml$break` in the way I do now.

File: R/get_syllabus.R
Comment (zkamvar) [inline R/get_syllabus.R:90]: > As the code highlighting suggested, `break` is a keyword in R. That means I cannot reference `my_yaml$break` in the way I do now.

That's on me, too! I should have seen that -_-. This will address that

```suggestion
  times <- c(coerce_integer(yaml$teaching), coerce_integer(yaml$exercises), coerce_integer(yaml[["break"]], 0L))
```

**Review by zkamvar — APPROVED** [review]

**Comment by zkamvar** [comment]
Thank you for this! I'm getting a green light on https://github.com/carpentries/sandpaper/actions/runs/4995718012/jobs/8948100405?pr=455, so that means that all the tests pass with the state of the world as it is today. 

> I had looked at the tests, but don't fully understand them yet – do they run in order and not reset to a default state between tests, or is there more magic going on with these unit tests?

Good question! The tests under `test/testthat/` are run in alphabetical order using the {testthat} package (see https://r-pkgs.org/testing-basics.html) via `devtools::test()` or `devtools::check()`. 

For this particular file, the tests _are_ dependent on one another (e.g. another test below this one needs to acknowledge that there is a break episode and adjust the expected timings accordingly). 

Because all of the tests need to work with a lesson, the first script to run is [`tests/testthat/setup.R`](https://github.com/carpentries/sandpaper/blob/6989c39f4482b51c22c8ceb38146d08ed8ca4069/tests/testthat/setup.R), where [a test lesson](https://carpentries.github.io/sandpaper/reference/fixtures.html) is created and stored in a temporary location for the duration of the test suite and a reset function is exposed for the tests. 

Each test file will reset the lesson and run the tests from top to bottom. In this way, the tests within a file are somewhat dependent on one another because they explicitly work on the lesson files. I have attempted to minimize this, but there are some times when the side-effects were necessary.

**Comment by zkamvar** [comment]
The checks are not passing, but that's because of #457, which has been fixed in main, so I'm going to merge this and then add the test and your contribution recognition, @bencomp 

Thank you!

**Pull request merged**
